app/models/host/base.rb

```diff
@@ -349,11 +349,13 @@ def get_interface_scope(name, attributes, base = self.interfaces)
           rescue Net::Validations::Error
             logger.debug "invalid mac during parsing: #{attributes[:macaddress]}"
           end
-          base = base.where(:mac => macaddress)
+
+          nics = base.where(:mac => macaddress)
```

Comment:
I think we do need to fallback to 'base.physical' if we can't find it by name, but i do like your structure better.  Updating!